### PR TITLE
Feature/added strategy

### PR DESCRIPTION
**Pull request by VallarDark**

### CODE REVIEW

File: OneSchedule.Domain.Models/Strategy/CreateStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using OneSchedule.Attributes;
+using System;
+using System.Threading.Tasks;
+
+namespace OneSchedule.Domain.Models.Strategy
+{
+    [StrategyName("create")]
+    public class CreateStrategy: IStrategy
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategy/CreateStrategy.cs]: Format code

File: OneSchedule.Domain.Models/Strategy/StrategyConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public static class StrategyConfiguration
+    {
+        public static IServiceCollection ConfigureStrategy(this IServiceCollection services)
+        {
+            services.AddSingleton<IStrategy, CreateStrategy>();
+            services.AddSingleton<IStrategy, EditStrategy>();
+            services.AddSingleton<IStrategy, FindStrategy>();
+            services.AddSingleton<IStrategy, RemoveStrategy>();
+
+            services.AddSingleton<StrategyContext>();
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategy/StrategyConfiguration.cs]: Why don't you use interface for a context?

File: OneSchedule.Domain.Models/Strategy/StrategyContext.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace OneSchedule.Domain.Models.Strategy
+{
+    public class StrategyContext
+    {
+        private readonly Dictionary<string, IStrategy> _strategies;
+
+        public StrategyContext(IEnumerable<IStrategy> strategies)
+        {
+            _strategies = new Dictionary<string, IStrategy>();
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategy/StrategyContext.cs]: You can try to use extension method ToDictionary() here.

File: OneSchedule/Controllers/UpdateController.cs [hunk 4]
```diff
@@ -37,9 +41,10 @@ public void Post([FromBody] Update[] updates)
         }
 
         [HttpGet]
-        public string Get()
+        public async Task<string> Get()
```
Comment (elkavitaly) [inline OneSchedule/Controllers/UpdateController.cs]: Why does Get method creates a new entity?

File: OneSchedule.Domain.Models/Strategy/CreateStrategy.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using OneSchedule.Attributes;
+using System;
+using System.Threading.Tasks;
+
+namespace OneSchedule.Domain.Models.Strategy
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategy/CreateStrategy.cs]: Rename folder from "Strategy" to "Strategies" and update namespaces accordingly

File: OneSchedule/Controllers/UpdateController.cs [hunk 6]
```diff
@@ -14,11 +16,13 @@ public class UpdateController : ControllerBase
 
         private readonly IService<UserDomain> _service;
         private readonly IMapper _mapper;
```
Comment (elkavitaly) [inline OneSchedule/Controllers/UpdateController.cs]: Are Service and Mapper used here?

**Review by elkavitaly — CHANGES_REQUESTED** [review]

File: OneSchedule.Domain.Models/Strategies/StrategyConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    public static class StrategyConfiguration
+    {
+        public static IServiceCollection ConfigureStrategy(this IServiceCollection services)
+        {
+            services.AddSingleton<IStrategy, CreateStrategy>();
+            services.AddSingleton<IStrategy, EditStrategy>();
+            services.AddSingleton<IStrategy, FindStrategy>();
+            services.AddSingleton<IStrategy, RemoveStrategy>();
+
+            services.AddSingleton<IStrategyContext,StrategyContext>();
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategies/StrategyConfiguration.cs]: Format code

File: OneSchedule.Domain.Models/Strategies/IStrategy.cs [hunk 8]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace OneSchedule.Domain.Models.Strategies
+{
+    public interface IStrategy
+    {
+        Task Execute(EventDomain eventDomain);
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategies/IStrategy.cs]: If the method is public and async use Async suffix in method's name, e.g. ExecuteAsync

File: OneSchedule.Domain.Models/Strategies/StrategyContext.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
+﻿using OneSchedule.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OneSchedule.Domain.Models.Strategies
+{
+    public class StrategyContext:IStrategyContext
+    {
```
Comment (elkavitaly) [inline OneSchedule.Domain.Models/Strategies/StrategyContext.cs:10]: Format code

File: OneSchedule.Helpers/CollectionNameReader.cs [hunk 10]
```diff
@@ -5,7 +5,7 @@ namespace OneSchedule.Helpers
 {
     public static class CollectionNameReader
     {
-        public static string GetName<T>()
+        public static string GetCollection<T>()
```
Comment (elkavitaly) [inline OneSchedule.Helpers/CollectionNameReader.cs]: It is better to name it GetCollectionName, because it doesn't retrieve any collection

**Review by elkavitaly — CHANGES_REQUESTED** [review]

**Review by elkavitaly — DISMISSED** [review]

**Review by gitrepository100 — DISMISSED** [review]
Approved

**Review by gitrepository100 — APPROVED** [review]
Approved

**Review by elkavitaly — APPROVED** [review]

**Pull request merged**